scene/2d/physics/collision_shape_2d.cpp

```diff
@@ -208,6 +208,24 @@ bool CollisionShape2D::is_disabled() const {
 	return disabled;
 }
 
+bool CollisionShape2D::contains_point(const Vector2 &p_point) const {
+	if (!collision_object)
+		return false;
+
+	Vector2 unscaledPoint;
+	Size2 scaleCheck;
+	if (collision_object->shape_owner_get_owner(owner_id)->is_class("Node2D")) {
+		Node2D *n = collision_object->shape_owner_get_owner(owner_id)->cast_to<Node2D>(collision_object->shape_owner_get_owner(owner_id));
+		unscaledPoint = (p_point - n->get_global_position()).rotated(n->get_global_rotation());
+		scaleCheck = n->get_global_scale();
+	} else {
+		Transform2D trans = collision_object->shape_owner_get_transform(owner_id);
+		unscaledPoint = (p_point - trans.get_origin()).rotated(trans.get_rotation());
+		scaleCheck = trans.get_scale();
+	}
+	return shape->contains_point(Vector2(unscaledPoint.x / scaleCheck.x, unscaledPoint.y / scaleCheck.y));
```

Comment:
```suggestion
	Vector2 unscaled_point;
	Size2 scale_check;
	if (Object::cast_to<Node2D>(collision_object->shape_owner_get_owner(owner_id)) {
		Node2D *n = Object::cast_to<Node2D>(collision_object->shape_owner_get_owner(owner_id));
		unscaled_point = (p_point - n->get_global_position()).rotated(n->get_global_rotation());
		scale_check = n->get_global_scale();
	} else {
		Transform2D trans = collision_object->shape_owner_get_transform(owner_id);
		unscaled_point = (p_point - trans.get_origin()).rotated(trans.get_rotation());
		scale_check = trans.get_scale();
	}
	return shape->contains_point(unscaled_point / scale_check);
```
Use `snake_case`, also you can operate directly on vectors, also use `cast_to` to check for type (it's a static function)

Reply:
Fixed by applying kleonc's fix to remove all potential variables.